pkg/controller/csi-driver-lvm/actuator.go

```diff
@@ -0,0 +1,788 @@
+package csidriverlvm
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gardener/gardener/extensions/pkg/controller/extension"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/metal-stack/gardener-extension-csi-driver-lvm/pkg/apis/config"
+	"github.com/metal-stack/gardener-extension-csi-driver-lvm/pkg/apis/csidriverlvm/v1alpha1"
+	"github.com/metal-stack/gardener-extension-csi-driver-lvm/pkg/imagevector"
+	"github.com/metal-stack/metal-lib/pkg/pointer"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	namespace string = "kube-system"
+
+	oldName        string = "csi-lvm"
+	oldNamespace   string = "csi-lvm"
+	oldProvisioner string = "metal-stack.io/csi-lvm"
+)
+
+// NewActuator returns an actuator responsible for Extension resources.
+func NewActuator(mgr manager.Manager, config config.ControllerConfiguration) extension.Actuator {
+	return &actuator{
+		client:  mgr.GetClient(),
+		decoder: serializer.NewCodecFactory(mgr.GetScheme(), serializer.EnableStrict).UniversalDecoder(),
+		config:  config,
+	}
+}
+
+type actuator struct {
+	client  client.Client
+	decoder runtime.Decoder
+	config  config.ControllerConfiguration
+}
+
+// Reconcile the Extension resource.
+func (a *actuator) Reconcile(ctx context.Context, log logr.Logger, ex *extensionsv1alpha1.Extension) error {
+	err := a.removeOldCsilvm(ctx)
+
+	if err != nil {
+		return fmt.Errorf("unable to remove old csi-lvm driver: %w", err)
+	}
+
+	log.Info("successfully removed old csi-lvm driver")
+
+	csidriverlvmConfig := &v1alpha1.CsiDriverLvmConfig{}
```

Comment:
Does it make sense to write a small validation for the configuration? You could check that the provided paths are absolute / contain a valid glob-pattern or something.